Border for `st.metric`

mayagbarnes opened the pull request:
## Describe your changes
PR adds `border` param for `st.metric` - default is `False` (current behavior)

## Testing Plan
- Unit Tests: ✅ 
- E2E Tests: ✅ 
- Manual Testing: ✅ 

-----
<img width="600" alt="Screenshot 2024-11-25 at 6 08 53 PM" src="https://github.com/user-attachments/assets/c8af802c-982c-40be-a676-094fa869528f">
<img width="600" alt="Screenshot 2024-11-25 at 6 08 44 PM" src="https://github.com/user-attachments/assets/9653e005-7a8e-40d6-af89-c847f3bf18fc">

mayagbarnes committed:
Add border param & tests

mayagbarnes committed:
Add snapshots

raethlein commented on frontend/lib/src/components/elements/Metric/Metric.tsx:
```diff
@@ -42,85 +41,74 @@ export interface MetricProps {
 
-  let direction: any = null
-  let color = ""
+  let metricDirection: any = null
```
Is it possible to have better typing here than `any`? It looks like it can be one of `ArrowDownward`, `ArrowUpward` or `null`

raethlein commented on frontend/lib/src/components/elements/Metric/Metric.tsx:
```diff
@@ -42,85 +41,74 @@ export interface MetricProps {
     // this must be none
     default:
-      direction = null
+      metricDirection = null
```
This does not seem to be required because you already initialize `metricDirection` with `null` in [line 56](https://github.com/streamlit/streamlit/pull/9928/files#diff-aa41fea932118f5233dbba41b388ea4465b8cd931325ba1e7297bb5c9affb35eR56)

raethlein commented on proto/streamlit/proto/Metric.proto:
```diff
@@ -29,6 +29,7 @@ message Metric {
   MetricColor color = 5;
   string help = 6;
   LabelVisibilityMessage label_visibility = 7;
+  bool border = 8;
```
Since this a boolean variable, I would name it something like `show_border` or something like this, also in the code. This variable name sounds more like the border definition itself.

sfc-gh-dmatthews commented on lib/streamlit/elements/metric.py:
```diff
@@ -112,6 +113,10 @@ def metric(
             If this is ``"collapsed"``, Streamlit displays no label or spacer.
 
+        border : bool
+            Whether to show a border around the metric container. If ``True``,
+            a border is shown. If ``False`` (default), no border is shown.
```
```suggestion
            Whether to show a border around the metric container. If this is
            ``False`` (default), no border is shown. If this is ``True``, a
            border is shown. 
```

mayagbarnes committed:
Address PR comments

mayagbarnes committed:
Merge branch 'develop' into metric-border

sfc-gh-dmatthews committed:
Update metric docstring (#9963)

## Describe your changes
Update metric docstring and add border exampel

---
**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this
project are made under the Apache 2.0 license.

wanjalaian commented:
So glad this is finally here
